docs/guides/socket-collector.md

```diff
@@ -0,0 +1,71 @@
+---
+title: 'The "socket-collector" gadget'
+weight: 10
+---
+
+socket-collector gathers information about TCP and UDP sockets. The "LOCAL" is
+the local IP address and port number pair. If connected, the "REMOTE" is the
+remote IP address and port number pair, otherwise, it will be "0.0.0.0:0". While
+"STATUS" is the internal status of the socket.
+
+We will start this demo by using nginx to create a web server on port 80:
+```
+$ kubectl create namespace test-socketcollector
+namespace/test-socketcollector created
+$ kubectl run --restart=Never --namespace test-socketcollector --image=nginx nginx-app --port=80
+pod/nginx-app created
+```
+
+Wait for the pod to get ready:
+```
+$ kubectl wait --timeout=-1s --namespace test-socketcollector --for=condition=ready pod/nginx-app ; kubectl get pod --namespace test-socketcollector
+pod/nginx-app condition met
+NAME        READY   STATUS    RESTARTS   AGE
+nginx-app   1/1     Running   0          46s
+```
+
+Use Inspektor Gadget to check the current TCP/UDP sockets table of the test-socketcollector namespace:
+```
+$ kubectl gadget socket-collector --namespace test-socketcollector
+NODE       NAMESPACE               POD          PROTOCOL    LOCAL         REMOTE       STATUS
+my-node    test-socketcollector    nginx-app    TCP         0.0.0.0:80    0.0.0.0:0    LISTEN
+```
+
+Consider that you can also filter by pod or label.
+
+Now, modify the nginx configuration to listen on port 8080 instead of 80 and reload the daemon:
+```
+$ kubectl exec --namespace test-socketcollector nginx-app -- /bin/bash -c "sed -i 's/listen \+80;/listen\t8080;/g' /etc/nginx/conf.d/default.conf && exec nginx -s reload"
+[...] signal process started
+```
+
+Check with Inspektor Gadget what are the sockets now:
+```
+$ kubectl gadget socket-collector --namespace test-socketcollector
+NODE       NAMESPACE               POD          PROTOCOL    LOCAL           REMOTE       STATUS
+my-node    test-socketcollector    nginx-app    TCP         0.0.0.0:8080    0.0.0.0:0    LISTEN
+```
+
+You can also use JSON format to display the information. Just take into account that IP address and port are displayed separated with this format:
```

Comment:
This sentence sounds a bit strange, because it's putting the action on the user, when the action is taken by the gadget... Maybe: "You can also get the information in JSON format, by passing the `--json` flag"?

Also, given that the tutorial starts with We, I think We would look nicer here.

Reply:
Agree.